### ISSUE TITLE
To many steps in GB from Mi Band 2

### ISSUE THREAD

**Issue by nazco** [Freeyourgadget/Gadgetbridge#1000]
#### Before opening an issue please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

---
#### Your issue is:
I have a new Mi Band 2 and noticed that the step count of the band ist quite accurate while the step count in GB is showing a bigger value (so it is not the Issue described in #449 ). After browsing the issues I found issue #685 which sounds similar but for another HW. As suggested in the issue I've checked the database export for multiple entries per minute. I found a lot of entries with a per second rate with the value -1 for all values so I think this happens while not wearing the band. But I also found some valid sample with differing rates (under 60 seconds) in the database. At this moment I'm not sure if the problem occurs on the hardware or in GB. But I've observed that if I try to only sync the band with the dedicated buttons and not by swiping down on the statistics activity the problem seems to happen less often. Maybe it happens when you try to update very fast. Maybe a problem with thread locking? I'm not that deep in android programming to check it in the code on my own.

#### Your wearable device is:

Mi Band 2 HW Rev. V0.16.2.2
Mi Band 2 SW Rev. 1.0.1.52

#### Your android version is:

Android 7.1.1

#### Your Gadgetbridge version is:

0.24.5 from F-Droid

**Comment by ashimokawa** [comment]
@nazco 

I don't think this is a locking problem, even if we transfer data twice, old data should be overwritten and there should be no duplicates due to the database schema.

I could imagine that is is a problem with live activity and/or periodic heart rate measurements. In these cases a lot of samples get written and I will check if we actually save the step data in these cases. We should not because they would land also in minute data and become duplicated.

**Comment by ashimokawa** [comment]
@nazco 
Most probably the data comes from the periodic heart rate measurements if you have them enabled or if you measure heart rate on the band manually.

Int that case when connected Gadgetbridge writes all the data to it's database.

@cpfeiffer 
Can we change the way we handle this? We are notified on the realtime data characteristics in case of manual hr measurements or periodic ones, and we listen always. How about enabling the ble notification only when actively turning on realtime activity in the charts activity tab? The HR data for periodic measurements or manual ones wont be lost, because they also land in minute data. This can also be tested with the Bip.

**Comment by nazco** [comment]
>Most probably the data comes from the periodic heart rate measurements if you have them enabled or if you measure heart rate on the band manually.

Yes, periodic heart rate is enabled and I've checked this yesterday too. But the problem is this istn't causing the problem. For periodic measurements (I've configured it to every 5 minutes) I see that everything works fine, every fifth minute entry contains a heart rate value. There are no per second values and same for manually triggert heart rate measures. There is a value for this minute but not a per second value.

**Comment by ashimokawa** [comment]
@nazco 
Not saying that is is the reason of this bug, but did you consider updating to the latest firmware?

**Comment by nazco** [comment]
I haven't updated yet but I have to correct my previous post. I've now tried to explicitly avoid the live tracking feature and there was no problem with the step count the last week. So it seems like the live tracking is causing this problem.
